django/contrib/auth/hashers.py

```diff
@@ -247,31 +262,43 @@ def encode(self, password, salt, iterations=None):
         hash = base64.b64encode(hash).decode('ascii').strip()
         return "%s$%d$%s$%s" % (self.algorithm, iterations, salt, hash)
 
-    def verify(self, password, encoded):
+    def decode(self, encoded):
         algorithm, iterations, salt, hash = encoded.split('$', 3)
         assert algorithm == self.algorithm
-        encoded_2 = self.encode(password, salt, int(iterations))
+        return {
+            'algorithm': algorithm,
+            'iterations': int(iterations),
+            'salt': salt,
+            'hash': hash,
+        }
+
+    def verify(self, password, encoded):
+        decoded_entry = self.decode(encoded)
+        assert decoded_entry['algorithm'] == self.algorithm
+        encoded_2 = self.encode(password,
+                                decoded_entry['salt'],
+                                decoded_entry['iterations'])
         return constant_time_compare(encoded, encoded_2)
 
     def safe_summary(self, encoded):
-        algorithm, iterations, salt, hash = encoded.split('$', 3)
-        assert algorithm == self.algorithm
+        decoded_entry = self.decode(encoded)
+        assert decoded_entry['algorithm'] == self.algorithm
         return {
-            _('algorithm'): algorithm,
-            _('iterations'): iterations,
-            _('salt'): mask_hash(salt),
-            _('hash'): mask_hash(hash),
+            _('algorithm'): decoded_entry['algorithm'],
+            _('iterations'): decoded_entry['iterations'],
+            _('salt'): mask_hash(decoded_entry['salt']),
+            _('hash'): mask_hash(decoded_entry['hash']),
         }
 
     def must_update(self, encoded):
-        algorithm, iterations, salt, hash = encoded.split('$', 3)
-        return int(iterations) != self.iterations
+        decoded_entry = self.decode(encoded)
+        return int(decoded_entry['iterations']) != self.iterations
```

Comment:
You can drop `int` here now (you already are doing it in `decode`), same for other hasher probably.

Reply:
Commit added to remove that and one more cast on the iterations value.